file/file_instance.go

```diff
@@ -72,11 +78,87 @@ func (f *fileInstance) GetDataRecords() []lib.Record {
 
 // GeneratorTrailer returns trailer segment that created automatically
 func (f *fileInstance) GeneratorTrailer() (lib.Record, error) {
-	return nil, nil
+	var trailer lib.Record
+	var information *lib.TrailerInformation
+	var err error
+	if f.format == PackedFileFormat {
+		trailer = lib.NewPackedTrailerRecord()
+		information, err = f.generatorPackedTrailer()
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		trailer = lib.NewTrailerRecord()
+		information, err = f.generatorTrailer()
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	fromFields := reflect.ValueOf(information).Elem()
+	toFields := reflect.ValueOf(trailer).Elem()
+	for i := 0; i < fromFields.NumField(); i++ {
+		fieldName := fromFields.Type().Field(i).Name
+		fromField := fromFields.FieldByName(fieldName)
+		toField := toFields.FieldByName(fieldName)
+		if fromField.IsValid() && toField.CanSet() {
+			toField.Set(fromField)
+		}
+	}
+
+	if f.format == PackedFileFormat {
+		segment := trailer.(*lib.PackedTrailerRecord)
```

Comment:
Can we check the type cast here? `segment, ok := trailer.(*lib.PackedTrailerRecord)`

Reply:
fixed